packages/x-date-pickers/src/DesktopTimePicker/DesktopTimePicker.tsx

```diff
@@ -18,6 +18,13 @@ const valueManager: PickerStateValueManager<unknown, unknown> = {
   parseInput: parsePickerInputValue,
   areValuesEqual: (utils: MuiPickersAdapter<unknown>, a: unknown, b: unknown) =>
     utils.isEqual(a, b),
+  updateValue: (utils, prevValue, newValue) => {
```

Comment:
Optional method (to be named in a better way) that creates the new value from the previous valud and the new one.

Allows us in TimePicker to only take the time from the new value but keep the date of the old value.

TODO: Do the same for MobileTimePicker (maybe refacto to  avoid duplication on the value managers at some point)

_____

date-fns handles this behavior out of the box, you can pass a reference date and when updating a mask it only applies the change of the elements in the mask, the rest is taken from the reference date.
It would be perfect here, but the others date libs do not support it from what I see.